### ISSUE TITLE
High power and mode switching

### ISSUE THREAD

**Issue by misterfish** [e-mo/rfm69_rp2040#3]
It seems the logic in rfm69_power_level_set and rfm69_mode_set may be admitting some erroneous edge cases.

For example, if you're already in RX mode, and then you set the power to max, then it will apply high power settings as far as I can see. (Haven't tested it).

**Comment by e-mo** [comment]
Correct. You have found an edge case. I intend to keep this library as zero assumption as possible, but that feels like something worth extracting away.

Should be an easy fix. Thanks for the find. Happy to have a second set of eyes.  I'll patch that shortly.

**Comment by e-mo** [comment]
I applied a bandaid fix to the problem, but it exposed a few places where the power routines can be simplified slightly. There isn't really any reason to actually set the device into high-power unless it is in TX mode. Thanks for pointing this out! I'm sure this would have created some devious bug that would have driven me closer to death somewhere down the line.

Anyway, the bandaid fix simply checks if the device is in RX mode and won't set high-power mode if that is the case. Should eliminate this edge case.
